docs/src/pages/components/data-grid/group-pivot/group-pivot.md

```diff
@@ -324,13 +324,114 @@ To do so, we need to override both the `renderCell` of the grouping column and t
 
 {{"demo": "pages/components/data-grid/group-pivot/TreeDataFullExample.js", "bg": "inline", "defaultCodeOpen": false}}
 
-## 🚧 Master detail [<span class="plan-pro"></span>](https://mui.com/store/items/material-ui-pro/)
+## Master detail [<span class="plan-pro"></span>](https://mui.com/store/items/material-ui-pro/)
 
-> ⚠️ This feature isn't implemented yet. It's coming.
+The master detail allows to expand a row to display additional information inside a panel.
+To start using this feature, pass a function to the `getDetailPanelContent` prop with the content to be rendered inside the panel.
+Any valid React element can be used as the row detail, even another grid.
+
+The height of the detail panel content needs to be provided upfront.
+The grid assumes the value of 500px by default.
+However, this can be configured by passing a function to the `getDetailPanelHeight` prop.
+Both props are called with a [`GridRowParams`](/api/data-grid/grid-row-params/) object, allowing to return a different value for each row.
+
+```tsx
+<DataGridPro
+  getDetailPanelContent={({ row }) => <div>Row ID: {row.id}</div>}
+  getDetailPanelHeight={({ row }) => 100} // Optional, default is 500px.
+/>
+```
+
+To expand a row, click in the "i" icon or press <kbd><kbd class="key">Ctrl</kbd>+<kbd class="key">Enter</kbd></kbd> inside one of the cells of the row.
+Returning `null` or `undefined` as the value of `getDetailPanelContent` will prevent the respective row from being expanded.
+
+{{"demo": "pages/components/data-grid/group-pivot/BasicDetailPanels.js", "bg": "inline", "defaultCodeOpen": false}}
+
+> ⚠ Always memoize the function provided to `getDetailPanelContent` and `getDetailPanelHeight`.
+> The grid bases on the referential value of these props to cache their values and optimize the rendering.
+>
+> ```tsx
+> const getDetailPanelContent = React.useCallback(() => { ... }, []);
 >
-> 👍 Upvote [issue #211](https://github.com/mui-org/material-ui-x/issues/211) if you want to see it land faster.
+> <DataGridPro getDetailPanelContent={getDetailPanelContent} />
+> ```
+>
+> ⚠ Depending on the height of the detail panel, you may see a blank space when scrolling.
+> This is caused because the grid uses a lazy approach to update the rendered rows.
+> Set `rowThreshold` to 0 to fix it:
+>
+> ```tsx
+> <DataGridPro rowThreshold={0} />
+> ```
+
+### Controlling the detail panels
+
+To control which rows are expanded, pass a list of row ids to the `detailPanelExpandedRowIds` prop.
+The `onDetailPanelExpandedRowIds` prop can be used to detect when a row is expanded or collapsed.
```

Comment:
> can be used to detect when a row is expanded

"is expanded", or "is requesting to be expanded"? Is this a controlled component, or is this a callback prop? (Asking because of the preceding use of "control".)

Reply:
I added more context about what is `onDetailPanelExpandedRowIds`.